File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:

        def sort(nums):
            N = len(nums)
            i = 0

            while(i < N):
                correct_index = nums[i] - 1

                if nums[correct_index] != correct_index + 1:
                    swap(nums, i, correct_index)
                else:
                    i += 1
            return nums
        

        def swap(nums, i, j):
            nums[i], nums[j] = nums[j], nums[i]
        
        nums = sort(nums)
        N = len(nums)
        ans = []

        for i in range(N):
            if nums[i] != i + 1:
                ans.append(i + 1)
        return ans
        

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        Set<Integer> set=new HashSet<>();
         List<Integer>  result=new ArrayList<>();
        for(int x:nums) set.add(x);
        int count=1;
        while(count<=nums.length){
            if(!set.contains(count)) 
              result.add(count);
            count++;
        }
        return result;
    }
}